File: fsharp/sublist/Sublist.fs
module Sublist

type Sublist = Equal | Sublist | Superlist | Unequal

let private isSublist l1 l2 = 
    List.isEmpty l1
    || (List.length l1 < List.length l2 
        && List.windowed l1.Length l2 |> List.contains l1)

let sublist xs ys = 
    if xs = ys then Equal
    elif isSublist ys xs then Superlist
    elif isSublist xs ys then Sublist
    else Unequal

File: go/sublist/sublist.go
package sublist

// Relation represents the relationship between two lists
type Relation string

// Sublist returns whether 'a' is a sublist, superlist, equal to or unequal to 'b'
func Sublist(a, b []int) Relation {
	if len(a) == len(b) && contains(a, b) {
		return "equal"
	}

	if len(a) < len(b) && contains(b, a) {
		return "sublist"
	}

	if len(b) < len(a) && contains(a, b) {
		return "superlist"
	}

	return "unequal"
}

func contains(parent, child []int) bool {
	for i := 0; i < len(parent)-len(child)+1; i++ {
		found := true
		for j := 0; j < len(child); j++ {
			if parent[i+j] != child[j] {
				found = false
				break
			}
		}
		if found {
			return true
		}
	}
	return false
}
